### ISSUE TITLE
Tolerate HTTP 504s during Storage Gateway creation

### ISSUE THREAD

**Issue by acdha** [terraform-providers/terraform-provider-aws#12618]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Description

When a Storage Gateway is being provisioned it may return HTTP 504s for a brief period while starting up. The step which retrieves the activation key should tolerate this.

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.55.0
+ provider.http v1.1.1
```

### Affected Resource(s)

* aws_storagegateway_gateway

### Terraform Configuration Files

```hcl
resource "aws_storagegateway_gateway" "app_storage" {
  gateway_ip_address       = aws_instance.sgw.public_dns
  gateway_name             = lower("${local.deployment_id}-sgw")
  gateway_timezone         = "GMT"
  gateway_type             = "FILE_S3"
  cloudwatch_log_group_arn = aws_cloudwatch_log_group.sgw.arn

  tags = merge(local.tags, {
    Name    = "${local.deployment_id} App Storage",
    Service = "Storage Gateway"
  })

  timeouts {
    create = "20m"
  }
}```

### Expected Behavior

The Storage Gateway would be polled until it returns 302 or times out

### Actual Behavior

```
Error: expected HTTP status code 302, received: 504

  on storagegateway.tf line 114, in resource "aws_storagegateway_gateway" "app_storage":
 114: resource "aws_storagegateway_gateway" "app_storage" {

```

### Steps to Reproduce

1. `terraform apply`

Running it a second time works so this is not a showstopper

**Comment by bflad** [comment]
Fix submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/12773

**Comment by bflad** [comment]
Support for this has been merged and will release with version 2.60.0 of the Terraform AWS Provider, later this week. 👍

**Comment by acdha** [comment]
Awesome, thanks!
